tests/httpserver.c

```diff
@@ -695,6 +695,46 @@ TestMain("HTTP Server", {
 		});
 	});
 
+	Convey("Multiple tree handlers works", {
+		char     urlstr[32];
+		nng_url *url;
+		char *   tmpdir;
+		char *   workdir;
+
+		trantest_next_address(urlstr, "http://127.0.0.1:%u");
+		So(nng_url_parse(&url, urlstr) == 0);
+		So(nng_http_server_hold(&s, url) == 0);
+		So((tmpdir = nni_plat_temp_dir()) != NULL);
+		So((workdir = nni_file_join(tmpdir, "httptest")) != NULL);
+
+		Reset({
+			nng_http_server_release(s);
+			free(tmpdir);
+			nni_file_delete(workdir);
+			free(workdir);
+			nng_url_free(url);
+		});
+
+		So(nng_http_handler_alloc_directory(&h, "/", workdir) == 0);
+		So(nng_http_server_add_handler(s, h) == 0);
+
+		So(nng_http_handler_alloc_directory(&h, "/", workdir) == 0);
+		So(nng_http_server_add_handler(s, h) == NNG_EADDRINUSE);
+		nng_http_handler_free(h);
+
+		So(nng_http_handler_alloc_directory(&h, "/subdir", workdir) ==
+		    0);
+		So(nng_http_server_add_handler(s, h) == 0);
+
+		So(nng_http_handler_alloc_directory(&h, "/subdir", workdir) ==
```

Comment:
I'd really like to have a test to ensure that the handler for the more specific one is called when a more specific path is provided.

Reply:
Yes, good point.